src/proj_equirect.jl

```diff
@@ -301,35 +274,83 @@ function *(M::BlockDiagEquiRect{QUAzFourier}, f::EquiRectQUAzFourier)
     EquiRectQUAzFourier(@tullio(Bf[p,iₘ] := M.blocks[p,q,iₘ] * f.arr[q,iₘ]), f.metadata)
 end
 
-# ## map, e.g. map((Bℓ,fℓ)->sqrt(Bℓ)*fℓ, BlkDiag, field) -> field
+# TODO: Figure out how reduce duplication so I can define methods like this ...
+
+# function *(M::BlockDiagEquiRect{T}, f::EquiRect{T}) where {T<:Az}
+# ...
+
+# ## mapblocks for fun.(Mblocks,eachcol(f))
+
+function mapblocks(fun::Function, M::BlockDiagEquiRect{B}, f::EquiRectField) where {B<:Basis} 
+    mapblocks(fun, M, B(f))
+end
 
-function Base.map(fun::Function, M::BlockDiagEquiRect{AzFourier}, f::F) where {F<:EquiRectAzFourier}
+function mapblocks(fun::Function, M::BlockDiagEquiRect{QUAzFourier}, f::EquiRectQUAzFourier)
     promote_metadata_strict(M.proj, f.proj) # ensure same projection
     Mfarr = similar(f.arr)
-    for (Mfc, Mc, fc) in zip(eachcol(Mfarr), M, eachcol(f.arr))
-        Mfc = fun(Mc, fc)
+    for i ∈ axes(M.blocks,3)
+        Mfarr[:,i] = fun(M.blocks[:,:,i], f.arr[:,i])
     end
-    EquiRectAzFourier(Mfarr, f.metadata)
+    EquiRectQUAzFourier(Mfarr, f.metadata)
 end 
 
-
-function Base.map(fun::Function, M::BlockDiagEquiRect{QUAzFourier}, f::F) where {F<:EquiRectQUAzFourier}
+function mapblocks(fun::Function, M::BlockDiagEquiRect{AzFourier}, f::EquiRectAzFourier)
     promote_metadata_strict(M.proj, f.proj) # ensure same projection
     Mfarr = similar(f.arr)
-    for (Mfc, Mc, fc) in zip(eachcol(Mfarr), M, eachcol(f.arr))
-        Mfc = fun(Mc, fc)
+    for i ∈ axes(M.blocks,3)
+        Mfarr[:,i] = fun(M.blocks[:,:,i], f.arr[:,i])
     end
```

Comment:
Can you just use `mapslices` here? I believe that has a fast GPU implementation, whereas this manual loop will not.